### PR TITLE
chore: refactor slog under emqx, conf and exhook

### PR DESCRIPTION
**Pull request by Spycsh**

### CODE REVIEW

**Comment by zhongwencool** [comment]
FYI: 
I seem to remember discussing SLOG planning before like this:
```erlang
#{msg => "RECV_packet", packet => Packet}
to 
#{"RECV_packet" => Packet}
```

**Comment by Spycsh** [comment]
> FYI: I seem to remember discussing SLOG planning before like this:
> 
> ```erlang
> #{msg => "RECV_packet", packet => Packet}
> to 
> #{"RECV_packet" => Packet}
> ```

Thanks!

File: apps/emqx_exhook/src/emqx_exhook_server.erl [hunk 1]
```diff
@@ -267,22 +269,23 @@ match_topic_filter(TopicName, TopicFilter) ->
...
-            ?LOG(error, "CALL ~0p:~0p(~0p, ~0p) error: ~0p",
-                        [?PB_CLIENT_MOD, Fun, Req, Options, Reason]),
+            ?SLOG(error, #{msg => "do_call_error", module => ?PB_CLIENT_MOD, function => Fun,
+                req => Req, options => Options, reason => Reason}),
             {error, Reason};
         {'EXIT', {Reason, Stk}} ->
-            ?LOG(error, "CALL ~0p:~0p(~0p, ~0p) throw an exception: ~0p, stacktrace: ~0p",
-                        [?PB_CLIENT_MOD, Fun, Req, Options, Reason, Stk]),
+            ?SLOG(error, #{msg => "do_call_exception", module => ?PB_CLIENT_MOD, function => Fun,
+                req => Req, options => Options, stk => Stk}),
```
Comment (zmstone) [inline apps/emqx_exhook/src/emqx_exhook_server.erl:288]:
```suggestion
                req => Req, options => Options, stacktrace => Stk}),
```

File: apps/emqx/src/emqx_session_router.erl [hunk 2]
```diff
@@ -215,7 +215,7 @@ handle_call({pending, SessionID, MarkerIDs}, _From, State) ->
     Res = emqx_persistent_session:pending_messages_in_db(SessionID, MarkerIDs),
     {reply, Res, State};
 handle_call(Req, _From, State) ->
-    ?LOG(error, "Unexpected call: ~p", [Req]),
+    ?SLOG(error, #{"unexpected_call" => Req}),
```
Comment (zmstone) [inline apps/emqx/src/emqx_session_router.erl:218]: msg field

File: apps/emqx/src/emqx_session_router.erl [hunk 3]
```diff
@@ -233,11 +233,11 @@ handle_cast({resume_end, SessionID, Pid}, State) ->
     _ = emqx_session_router_worker_sup:abort_worker(Pid),
     {noreply, State#{ pmon => Pmon }};
 handle_cast(Msg, State) ->
-    ?LOG(error, "Unexpected cast: ~p", [Msg]),
+    ?SLOG(error, #{"unexpected_cast" => Msg}),
```
Comment (zmstone) [inline apps/emqx/src/emqx_session_router.erl:236]: msg field

File: apps/emqx/src/emqx_session_router.erl [hunk 4]
```diff
@@ -233,11 +233,11 @@ handle_cast({resume_end, SessionID, Pid}, State) ->
...
     {noreply, State}.
 
 handle_info(Info, State) ->
-    ?LOG(error, "Unexpected info: ~p", [Info]),
+    ?SLOG(error, #{"unexpected_info" => Info}),
```
Comment (zmstone) [inline apps/emqx/src/emqx_session_router.erl:240]: msg field

File: apps/emqx_conf/src/emqx_cluster_rpc_handler.erl [hunk 5]
```diff
@@ -41,22 +41,22 @@ init([State]) ->
     {ok, ensure_timer(State)}.
 
 handle_call(Req, _From, State) ->
-    ?LOG(error, "unexpected call: ~p", [Req]),
+    ?SLOG(error, #{"unexpected_call" => Req}),
```
Comment (zmstone) [inline apps/emqx_conf/src/emqx_cluster_rpc_handler.erl:44]: msg field

File: apps/emqx_conf/src/emqx_cluster_rpc_handler.erl [hunk 6]
```diff
@@ -41,22 +41,22 @@ init([State]) ->
...
     {reply, ignored, State}.
 
 handle_cast(Msg, State) ->
-    ?LOG(error, "unexpected msg: ~p", [Msg]),
+    ?SLOG(error, #{"unexpected_msg" => Msg}),
```
Comment (zmstone) [inline apps/emqx_conf/src/emqx_cluster_rpc_handler.erl:48]: msg

File: apps/emqx_conf/src/emqx_cluster_rpc_handler.erl [hunk 7]
```diff
@@ -41,22 +41,22 @@ init([State]) ->
...
     {noreply, State}.
 
 handle_info({timeout, TRef, del_stale_mfa}, State = #{timer := TRef, max_history := MaxHistory}) ->
     case mria:transaction(?CLUSTER_RPC_SHARD, fun del_stale_mfa/1, [MaxHistory]) of
         {atomic, ok} -> ok;
-        Error -> ?LOG(error, "del_stale_cluster_rpc_mfa error:~p", [Error])
+        Error -> ?SLOG(error, #{msg => "del_stale_cluster_rpc_mfa_error", error => Error})
```
Comment (zmstone) [inline apps/emqx_conf/src/emqx_cluster_rpc_handler.erl:54]: msg

File: apps/emqx_conf/src/emqx_cluster_rpc_handler.erl [hunk 8]
```diff
@@ -41,22 +41,22 @@ init([State]) ->
...
     end,
     {noreply, ensure_timer(State), hibernate};
 
 handle_info(Info, State) ->
-    ?LOG(error, "unexpected info: ~p", [Info]),
+    ?SLOG(error, #{"unexpected_info" => Info}),
```
Comment (zmstone) [inline apps/emqx_conf/src/emqx_cluster_rpc_handler.erl:59]: msg

File: apps/emqx_exhook/src/emqx_exhook_server.erl [hunk 9]
```diff
@@ -155,8 +155,10 @@ do_init(ChannName, ReqOpts) ->
             try
                 {ok, resolve_hookspec(maps:get(hooks, InitialResp, []))}
             catch _:Reason:Stk ->
-                ?LOG(error, "try to init ~p failed, reason: ~p, stacktrace: ~0p",
-                             [ChannName, Reason, Stk]),
+                ?SLOG(error, #{msg => "failed_to_init_channel_failed",
```
Comment (zmstone) [inline apps/emqx_exhook/src/emqx_exhook_server.erl:158]:
```suggestion
                ?SLOG(error, #{msg => "failed_to_init_channel",
```

**Review by zmstone — APPROVED** [review]

**Pull request merged**